src/pages/studyView/resources/FilesAndLinks.tsx

```diff
@@ -0,0 +1,329 @@
+import * as React from 'react';
+import {
+    Column,
+    default as LazyMobXTable,
+} from 'shared/components/lazyMobXTable/LazyMobXTable';
+import { observer } from 'mobx-react';
+import _, { forEach } from 'lodash';
+import internalClient from 'shared/api/cbioportalInternalClientInstance';
+import { Else, If, Then } from 'react-if';
+import { WindowWidthBox } from '../../../shared/components/WindowWidthBox/WindowWidthBox';
+import LoadingIndicator from 'shared/components/loadingIndicator/LoadingIndicator';
+import {
+    getSampleViewUrlWithPathname,
+    getPatientViewUrlWithPathname,
+} from 'shared/api/urls';
+import { getAllClinicalDataByStudyViewFilter } from '../StudyViewUtils';
+import { StudyViewPageStore } from 'pages/studyView/StudyViewPageStore';
+import { isUrl, remoteData } from 'cbioportal-frontend-commons';
+import { makeObservable, observable, computed } from 'mobx';
+import { ResourceData, StudyViewFilter } from 'cbioportal-ts-api-client';
+
+export interface IFilesLinksTable {
+    store: StudyViewPageStore;
+}
+
+class FilesLinksTableComponent extends LazyMobXTable<{
+    [id: string]: string;
+}> {}
+
+const RECORD_LIMIT = 500;
+
+async function fetchResourceDataOfPatient(patientIds: Map<string, string>) {
+    const ret: { [key: string]: ResourceData[] } = {};
+    const promises = [];
+    for (let [patientId, studyId] of patientIds) {
+        promises.push(
+            internalClient
+                .getAllResourceDataOfPatientInStudyUsingGET({
+                    studyId: studyId,
+                    patientId: patientId,
+                    projection: 'DETAILED',
+                })
+                .then(data => {
+                    if (patientId in ret) {
+                        ret[patientId].push(...data);
+                    } else {
+                        ret[patientId] = data;
+                    }
+                })
+        );
+    }
+
+    return Promise.all(promises).then(() => ret);
+}
+
+async function fetchFilesLinksData(
+    filters: StudyViewFilter,
+    sampleIdResourceData: { [sampleId: string]: ResourceData[] },
+    searchTerm: string | undefined,
+    sortAttributeId: string | undefined,
+    sortDirection: 'asc' | 'desc' | undefined,
+    recordLimit: number
+) {
+    let sampleClinicalDataResponse = await getAllClinicalDataByStudyViewFilter(
+        filters,
+        searchTerm,
+        sortAttributeId,
+        sortDirection,
+        recordLimit,
+        0
+    );
+
+    const patientIds = new Map<string, string>();
+
+    forEach(sampleClinicalDataResponse.data, (data, uniqueSampleId) => {
+        forEach(data, item => {
+            patientIds.set(item.patientId, item.studyId);
+        });
+    });
+
+    const resourcesPerPatient: { [key: string]: number } = {};
+    const items: { [attributeId: string]: any } = [];
```

Comment:
this type doesn't see right.  it's type as a map, but we assign an array to it.

Reply:
also, lets see if we can improve on "any".   What does the table accept?

Reply:
string and numbers (due to sorting of resourcesPerPatient), do you want me to cast number to string? And then we can change any to string.

I saw some parsing in getDefaultColumnConfig -> sortBy, so it might work.